aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -134,6 +136,15 @@ module aptos_framework::block {
         };
         emit_new_block_event(&vm, &mut block_metadata_ref.new_block_events, new_block_event);
 
+        if (features::collect_and_distribute_gas_fees()) {
+            // Assign the fees collected from the previous block to the previous block proposer.
+            // If for any reason the fees cannot be assigned, this function burns the collected coins.
+            transaction_fee::process_collected_fees();
+            // Set the proposer of this block as the receiver of the fees, so that the fees for this
+            // block are assigned to the right account.
+            transaction_fee::register_proposer_for_fee_collection(proposer);
```

Comment:
we should filter out the propose with @vm_reserved address

Reply:
I guess you mean `register_proposer_for_fee_collection`, right?